### PR TITLE
Improved wiring support and updated cookie instructions

### PR DESCRIPTION
**Pull request by KeltenP**
I found this repo last week, and upon trying it out I found the manual was working fine but the wiring was not working at all. After reading some of the issues, looking at the open (but stale) pull request, and taking a look at some of the forks, I took a stab at getting this into decent working condition again.

These changes improves wiring downloads, including:
- Downloads all pages for all wiring sections
- Downloads all connector views
- Direct download of svg files (converted to png) for "Page" type documents, instead of doing a screenshot of the page. The screenshot was being cutoff if the wiring page was too large. This preserves the entirety of the content.
- Adds missing parameters to some requests (e.g. languageCode). These requests were failing without these parameters.

I was also having issues with cookies and seemingly not having the right ones. I've updated the instructions in the README file for retrieving additional cookies and this seems to be working well.

Using these updates I was able to download the full manual, all wiring (including every page for every wiring section) and all connector views for the following vehicles:
- 2022 Ford Escape ("Page" type)
- 2008 Ford Focus ("Page" type)
- 2004 Ford Focus ("BasicPage" type)

Unfortunately I ran out of time to test any older vehicles (e.g. pre-2003). I did add some TODOs into the code for some obvious changes that need to be made to get those working.

This is my first time working in TypeScript/nodejs. So hopefully this code is mostly up to snuff :)

### CODE REVIEW

**Comment by buckchoklit** [comment]
@KeltenP Amazing work! I grabbed the code from this PR and was able to get most of the wiring diagrams loaded. However, it seems to be having issues with the connectors on a 2017 F150.

**Comment by iamtheyammer** [comment]
Hey, thanks so much for this awesome PR. Will check it out and get it merged this weekend. Really appreciate your hard work!

**Comment by KeltenP** [comment]
@buckchoklit do you happen to have the terminal output of the failure? 

Could you open the toc.json file in your Wiring folder, find the entry for the connectors (likely number 150) and post it here?

**Comment by buckchoklit** [comment]
I don't have terminal output any longer, as I was running this on an Ubuntu
VM that's already shut down. My subscription also ran out.

However, the error was net::ERR_HTTP2_PROTOCOL_ERROR. I believe this is a
server side issue caused by IIS based on this stack overflow article.

https://stackoverflow.com/questions/58215104/whats-the-neterr-http2-protocol-error-about

Basically IIS on the server side has a setting called "minBytesPerSecond"
that cuts the connection because it thinks its a "slow drip" DDOS.

I was able to pull down the connectors for a 2003 F150 with no issue.
However, the vehicles I looked at 2014+ all had issues.

File: src/wiring/saveEntireWiring.ts [hunk 1]
```diff
@@ -27,56 +31,204 @@ export default async function saveEntireWiring(
...
+        const viewBox = svgParsed.getElementById("svgDoc").getAttribute("viewBox");
+        // viewbox is "x y width height"
+        const width = viewBox.split(' ')[2];
+        const height = viewBox.split(' ')[3];
+        const image = await svgToImg.from(svgData).toPng({
+          path: newTitle,
+          background: 'white',
+          height: height,
+          width: width
+        });
```
Comment (iamtheyammer) [inline src/wiring/saveEntireWiring.ts:113]: I'm almost certain that we can do this inside Playwright: if we just call [page.setContent](https://playwright.dev/docs/api/class-page#page-set-content) on the SVG, that'll load it up in the browser. Then, we can query the browser for these SVG properties (if needed) and PDF it right from there. PDFing it also keeps them as vectors.

File: src/wiring/saveEntireWiring.ts [hunk 2]
```diff
@@ -27,56 +31,204 @@ export default async function saveEntireWiring(
...
+        const sectionPath = join(wiringPath, title);
+        try {
+          await mkdir(sectionPath);
+        } catch (e: any) {
+          if (e.code !== "EEXIST") {
+            throw e;
+          }
+        }
+        console.log(
+          `Downloading wiring diagram Section: ${doc.Number} Title: ${doc.Title} Page: ${page}. `
```
Comment (iamtheyammer) [inline src/wiring/saveEntireWiring.ts:166]:
```suggestion
          `Downloading wiring diagram: ${doc.Number} Title: ${doc.Title} Page: ${page}.`
```

File: src/wiring/saveEntireWiring.ts [hunk 3]
```diff
@@ -27,56 +31,204 @@ export default async function saveEntireWiring(
...
-      join(
-        wiringPath,
-        `${sanitizedTitle} (${doc.Filename.slice(
-          0,
-          doc.Filename.indexOf(".")
-        )})${doc.Filename.slice(doc.Filename.indexOf("."))}`
-      )
-    );
+    // Start connectors
+    if (doc.Type === "Connectors" && doc.Title === "Connector Views") {
```
Comment (iamtheyammer) [inline src/wiring/saveEntireWiring.ts:198]: Looks like we might be able to combine all these for loops into one by fetching all these lists at once then looping through every document switching on its type/title.

File: templates/cookieString.txt.template [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+	dtCookie
+	Ford.TSO.PTSSuite
+	TPS%2DPERM
+	PREFERENCES
+	TPS%2DMEMBERSHIP
+	AKA_A2
+	bm_mi
+	bm_sv
+	PTSSession
+4. Ensure the cookie string does not contain the ak_bmsc cookie
```
Comment (iamtheyammer) [inline templates/cookieString.txt.template:15]:
```suggestion
```

As stated above, we can programmatically remove this in the cookie parser

**Review by iamtheyammer — COMMENTED** [review]
Thanks again for this awesome PR. I'm leaving these review comments mostly for me as a to-do list. There's definitely some code cleanliness stuff that I can work on, but the hard stuff is done, thanks to you!

**Comment by iamtheyammer** [comment]
Got a PTS login from @strictlysimple, so I've modified your work and it's ready the lovely folks in #21 to test out.

Merging into `experimental-wiring` in case there are bugs as they run it, then I'll merge into `main`.

Thanks again to @KeltenP for this awesome PR!

**Pull request merged**